sdk/component/store/remote/findings-client/client_test.go

```diff
@@ -0,0 +1,187 @@
+package findingsclient_test
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	"github.com/smithy-security/smithy/sdk/component/store"
+	findingsclient "github.com/smithy-security/smithy/sdk/component/store/remote/findings-client"
+	"github.com/smithy-security/smithy/sdk/component/uuid"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	v1 "github.com/smithy-security/smithy/sdk/gen/findings_service/v1"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+type (
+	FindingsClientTestSuite struct {
+		suite.Suite
+
+		findingsServiceClient component.Storer
+		fakeFindingsService   fakeFindingsService
+	}
+
+	fakeFindingsService struct {
+		grpcServer *grpc.Server
+		v1.UnimplementedFindingsServiceServer
+		findings map[string]map[uint64]*ocsf.VulnerabilityFinding
+	}
+)
+
+func (ffs *fakeFindingsService) GetFindings(_ context.Context, req *v1.GetFindingsRequest) (*v1.GetFindingsResponse, error) {
+	if req == nil {
+		return &v1.GetFindingsResponse{}, errors.New("create findings request is nil")
+	}
+
+	findings, ok := ffs.findings[req.Id]
+	if !ok {
+		return &v1.GetFindingsResponse{}, status.Errorf(codes.NotFound, "findings not found")
+	}
+
+	var resFindings = make([]*v1.Finding, 0, len(findings))
+	for id, finding := range findings {
+		resFindings = append(resFindings, &v1.Finding{
+			Id:      id,
+			Details: finding,
+		})
+	}
+
+	return &v1.GetFindingsResponse{
+		Findings: resFindings,
+	}, nil
+}
+
+func (ffs *fakeFindingsService) UpdateFindings(_ context.Context, req *v1.UpdateFindingsRequest) (*v1.UpdateFindingsResponse, error) {
+	if req == nil {
+		return &v1.UpdateFindingsResponse{}, errors.New("create findings request is nil")
+	}
+
+	_, ok := ffs.findings[req.Id]
+	if !ok {
+		return &v1.UpdateFindingsResponse{}, status.Errorf(codes.NotFound, "findings not found")
+	}
+
+	for _, finding := range req.Findings {
+		if _, ok := ffs.findings[req.Id]; !ok {
+			ffs.findings[req.Id] = make(map[uint64]*ocsf.VulnerabilityFinding)
+		}
+		ffs.findings[req.Id][finding.Id] = finding.Details
+	}
+
+	return &v1.UpdateFindingsResponse{}, nil
+}
+
+func (ffs *fakeFindingsService) CreateFindings(_ context.Context, req *v1.CreateFindingsRequest) (*v1.CreateFindingsResponse, error) {
+	if req == nil {
+		return &v1.CreateFindingsResponse{}, errors.New("create findings request is nil")
+	}
+
+	if ffs.findings == nil {
+		ffs.findings = make(map[string]map[uint64]*ocsf.VulnerabilityFinding)
```

Comment:
Could you please add a small comment here to say what that means? `make(map[string]map[uint64]*ocsf.VulnerabilityFinding)` . Same with further down at `ffs.findings[req.Id][uint64(len(existingFindings)+1)] = finding`

Reply:
Done